src/Fluid/CPU_FluidSolver.cpp

```diff
@@ -250,22 +282,52 @@ void CPU_FluidSolver( real h_Flu_Array_In[][FLU_NIN][ CUBE(FLU_NXT) ],
 
 #  elif ( MODEL == ELBDM )
 
-#  if ( WAVE_SCHEME == WAVE_FD )
 
-// evaluate the optimized Taylor expansion coefficient
-   if ( ELBDM_Taylor3_Auto )  ELBDM_Taylor3_Coeff = ELBDM_SetTaylor3Coeff( dt, dh, ELBDM_Eta );
+#  if ( ELBDM_SCHEME == ELBDM_HYBRID )
+   if ( UseWaveFlag ) {
+#  endif // # if ( ELBDM_SCHEME == ELBDM_HYBRID )
 
+#  if (WAVE_SCHEME == WAVE_FD )
+//    evaluate the optimized Taylor expansion coefficient
+   if ( ELBDM_Taylor3_Auto )  ELBDM_Taylor3_Coeff = ELBDM_SetTaylor3Coeff( dt, dh, ELBDM_Eta );
    CPU_ELBDMSolver( h_Flu_Array_In, h_Flu_Array_Out, h_Flux_Array, NPatchGroup, dt, dh, ELBDM_Eta, StoreFlux,
                      ELBDM_Taylor3_Coeff, XYZ, MinDens );
 
 #  elif ( WAVE_SCHEME == WAVE_GRAMFE ) // #  if (WAVE_SCHEME == WAVE_FD )
 
-   CPU_ELBDMSolver_GramFE( h_Flu_Array_In, h_Flu_Array_Out, h_Flux_Array, NPatchGroup, dt, dh, ELBDM_Eta, StoreFlux,
+#  if ( GRAMFE_SCHEME == GRAMFE_FFT )
+   CPU_ELBDMSolver_GramFE_FFT( h_Flu_Array_In, h_Flu_Array_Out, h_Flux_Array, NPatchGroup, dt, dh, ELBDM_Eta, StoreFlux,
+                     XYZ, MinDens );
+#  elif ( GRAMFE_SCHEME == GRAMFE_MATMUL )
+// evaluate time evolution matrix
+   ELBDM_GramFE_ComputeTimeEvolutionMatrix(h_GramFE_TimeEvo, dt, dh, ELBDM_Eta);
```

Comment:
@hyschive 
I have addressed your comment about about `(*A[2])`, for the `HasWaveCounterpart` and `IsCompletelyRefined` arrays. 
However, I have left `h_GramFE_TimeEvo` as it is because it only needs to be computed once per time step and not on a patch by patch basis.